src/hugr.rs

```diff
@@ -150,13 +150,9 @@ impl NodeType {
     }
 }
 
-impl OpType {
-    /// Convert an OpType to a NodeType by giving it some input extensions
-    pub fn with_extensions(self, rs: ExtensionSet) -> NodeType {
-        NodeType {
-            op: self,
-            input_extensions: Some(rs),
-        }
+impl<T: Into<OpType>> From<T> for NodeType {
```

Comment:
Funky.

This may restrict some other impls for things that don't want the transitiveness. But it should be alright as long as `new_auto` is the sane default.